File: backend/app/models/product.py
from sqlmodel import Field, SQLModel, Relationship
from typing import List, Optional
from models.base_model import BaseModel

class Product(BaseModel, table=True):
    id: int = Field(default=None, primary_key=True)
    base_price: float
    name: str
    furniture_type: str
    product_type: str
    height: float
    width: float
    depth: float
    image_url: Optional[str] = None
    variances: List["Variance"] = Relationship(back_populates="product", cascade_delete=True)

File: frontend/vue-project/src/models/Product.ts
import type { Variance } from "./Variance"

export interface Product {
    id: number
    name: string
    base_price: number
    furniture_type: string
    product_type: string
    height: number
    width: number
    depth: number
    variances: Variance[]
}

export const formatPrice = (price: number): string => "$" + price.toFixed(2)